OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -468,11 +481,37 @@ bool CancelProductionInner(string itemName)
 		public void EndProduction(ProductionItem item)
 		{
 			Queue.Remove(item);
+
+			if (item.Infinite)
+			{
+				item.Reset();
+				Queue.Add(item);
+			}
 		}
 
 		protected void BeginProduction(ProductionItem item)
 		{
+			if (Queue.Any(i => i.Item.Equals(item.Item) && i.Infinite))
```

Comment:
ProductionItem doesn't override `.Equals`, so isn't this just doing a reference comparison?
This would make the whole line equivalent to `i.Infinite && Queue.Contains(i)`, which i don't think is what we want.

Reply:
i.Item is the Item string in ProductionItem